source/pool/pool.go

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+var WithReplCtxKey = struct{}{}
```

Comment:
I'd make this a bit safer and easier to use by hiding the key and exposing a method instead.

```suggestion
type replicationCtxKey struct{}

func WithReplication(ctx context.Context) context.Context {
	return context.WithValue(ctx, replicationCtxKey{}, true)
}
```